metals/src/main/scala/scala/meta/internal/metals/debug/BuildTargetClasses.scala

```diff
@@ -6,23 +6,41 @@ import scala.concurrent.Future
 
 import scala.meta.internal.metals.BatchedFunction
 import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.Compilers
 import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.SemanticdbFeatureProvider
 import scala.meta.internal.metals.debug.BuildTargetClasses.Classes
 import scala.meta.internal.metals.debug.BuildTargetClasses.TestSymbolInfo
+import scala.meta.internal.pc.PcSymbolInformation
+import scala.meta.internal.semanticdb.ClassSignature
 import scala.meta.internal.semanticdb.Scala.Descriptor
 import scala.meta.internal.semanticdb.Scala.Symbols
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.semanticdb.TextDocuments
+import scala.meta.internal.semanticdb.TypeRef
+import scala.meta.io.AbsolutePath
 
 import bloop.config.Config.TestFramework
 import ch.epfl.scala.{bsp4j => b}
 
 /**
  * In-memory index of main class symbols grouped by their enclosing build target
  */
-final class BuildTargetClasses(val buildTargets: BuildTargets)(implicit
+final class BuildTargetClasses(
+    val buildTargets: BuildTargets,
+    val compilers: () => Compilers,
+)(implicit
     val ec: ExecutionContext
-) {
+) extends SemanticdbFeatureProvider {
   private val index = TrieMap.empty[b.BuildTargetIdentifier, Classes]
 
+  private val bazelTestClassCache =
+    TrieMap.empty[AbsolutePath, List[(String, TestSymbolInfo)]]
+
+  private val symbolInfoCache =
```

Comment:
We might need a second cache to be efficient:
```
private val  symbolDefinitionCache = TrieMap[AbsolutePath, Set[String]]
```
then on change to any AbsolutePath we can just remove everything from the symbolInfoCache map.

Otherwise we need to go into every symbol of the test document.